_posts/2021-04-12-ballista-donation.md

```diff
@@ -0,0 +1,75 @@
+date: "2021-04-12 00:00:00 -0600"
+author: agrove
+categories: [application]
+---
+<!--
+{% comment %}
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to you under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+We are excited to announce that [Ballista](https://github.com/apache/arrow/tree/master/rust/ballista) has been donated 
+to the Apache Arrow project. 
+
+Ballista is a distributed compute platform primarily implemented in Rust, and powered by Apache Arrow. It is built
+on an architecture that allows other programming languages (such as Python, C++, and Java) to be supported as
+first-class citizens without paying a penalty for serialization costs.
+
+The foundational technologies in Ballista are:
+
+- [Apache Arrow](https://arrow.apache.org/) memory model and compute kernels for efficient processing of data.
+- [Apache Arrow Flight Protocol](https://arrow.apache.org/blog/2019/10/13/introducing-arrow-flight/) for efficient
+  data transfer between processes.
+- [Google Protocol Buffers](https://developers.google.com/protocol-buffers) for serializing query plans.
+- [Docker](https://www.docker.com/) for packaging up executors along with user-defined code.
+
+Ballista can be deployed as a standalone cluster and also supports [Kubernetes](https://kubernetes.io/). In either
+case, the scheduler can be configured to use [etcd](https://etcd.io/) as a backing store to (eventually) provide
+redundancy in the case of a scheduler failing.
+
+## Status
+
+The Ballista project is at an early stage of development. However, it is capable of running complex analytics queries 
+in a distributed cluster with reasonable performance.
+
+The following chart shows the current performance of Ballista for a number of TPC-H queries at scale factor 100. The 
+current performance at this scale is similar to Apache Spark, and better in some cases. More work is now needed to 
+make Ballista scale with larger data sets.
+
+![Ballista Performance](img/20210412ballistatpch.png)
+
+_Benchmarks were executed on a 24 core desktop with 64 GB RAM and NVMe drives. The 100 GB dataset was converted to 
+Parquet and repartitioned with 8 partitions._
```

Comment:
To be gracious, you might point out that these benchmarks haven't been peer reviewed and you look forward to working with the Spark community to create realistic and reproducible benchmarks, since the benchmarks are more about learning (so you can make the software better) rather than trying to show "which is better".